### PR TITLE
V8: add `FromValue`, error traits, roundtrip tests

### PR DESCRIPTION
**Pull request by Centril**
# Description of Changes

The primary goal here is to add the trait `FromValue` and its implementation.
These are then tested in combination with `ToValue`.
Some error handling functionality is also added.

# API and ABI breaking changes

None

# Expected complexity level and risk

1

# Testing

Proptests and edge case tests are added, testing that a value `x: T` is the same as `from_value(to_value(x))`.

### CODE REVIEW

File: crates/core/src/host/v8/from_value.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+int32_from_value!(Uint32, u32);
+int32_from_value!(Int32, i8);
+int32_from_value!(Int32, i16);
+int32_from_value!(Int32, i32);
+
+// `FromValue for f32, f64`.
+macro_rules! float_from_value {
+    ($rust_ty:ty) => {
+        impl_from_value!($rust_ty, (val, scope) => {
+            cast!(scope, val, Number, "number for `{}`", stringify!($rust_ty)).map(|n| n.value() as _)
```
Comment (gefjon) [inline crates/core/src/host/v8/from_value.rs:85]: What will this do for `f32` when the source value has more significant bits in its mantissa than `f32` can hold? Truncate? Round? Error?

Comment (Centril) [reply crates/core/src/host/v8/from_value.rs:85]: This is defined here: https://doc.rust-lang.org/reference/expressions/operator-expr.html#r-expr.as.numeric.float-narrowing

> Casting from an f64 to an f32 will produce the closest possible f32
> - if necessary, rounding is according to roundTiesToEven mode ***
>  - on overflow, infinity (of the same sign as the input) is produced
>
> *** as defined in IEEE 754-2008 §4.3.1: pick the nearest floating point number, preferring the one with an even least significant digit if exactly halfway between two floating point numbers.

I've left a note in the code.

**Review by gefjon — APPROVED** [review]
